src/dbus_api/pool/pool_3_8/methods.rs

```diff
@@ -0,0 +1,399 @@
+            return_message.append3(false, DbusErrorEnum::OK as u16, OK_STRING.to_string())
+        }
+        Ok(CreateAction::Created(_)) => {
+            dbus_context.push_pool_clevis_info_change(
+                pool_path.get_name(),
+                pool.encryption_info()
+                    .map(|either| either.map_left(|ei| (lowest_token_slot.is_none(), ei))),
+            );
+            return_message.append3(true, DbusErrorEnum::OK as u16, OK_STRING.to_string())
+        }
+        Err(e) => {
+            let (rc, rs) = engine_to_dbus_err_tuple(&e);
+            return_message.append3(default_return, rc, rs)
+        }
+    };
+    Ok(vec![msg])
+}
+
+pub fn unbind_clevis(m: &MethodInfo<'_, MTSync<TData>, TData>) -> MethodResult {
+    let message: &Message = m.msg;
+
+    let mut iter = message.iter_init();
+    let token_slot_tuple: (bool, u32) = get_next_arg(&mut iter, 0)?;
+
+    let dbus_context = m.tree.get_data();
+    let object_path = m.path.get_name();
+    let return_message = message.method_return();
+    let default_return = false;
+
+    let token_slot = tuple_to_option(token_slot_tuple);
+
+    let pool_path = m
+        .tree
+        .get(object_path)
+        .expect("implicit argument must be in tree");
+    let pool_uuid = typed_uuid!(
+        get_data!(pool_path; default_return; return_message).uuid;
+        Pool;
+        default_return;
+        return_message
+    );
+
+    let mut pool = get_mut_pool!(dbus_context.engine; pool_uuid; default_return; return_message);
+    let lowest_token_slot = pool
+        .encryption_info()
+        .and_then(|either| either.left())
+        .as_ref()
+        .and_then(|ei| ei.single_clevis_info())
+        .map(|(token_slot, _)| token_slot);
+
+    let msg = match handle_action!(
+        pool.unbind_clevis(token_slot),
+        dbus_context,
+        pool_path.get_name()
+    ) {
+        Ok(DeleteAction::Identity) => {
+            return_message.append3(false, DbusErrorEnum::OK as u16, OK_STRING.to_string())
+        }
+        Ok(DeleteAction::Deleted(_)) => {
+            dbus_context.push_pool_clevis_info_change(
+                pool_path.get_name(),
+                pool.encryption_info().map(|either| {
+                    either.map_left(|ei| {
+                        (token_slot.is_some() && token_slot == lowest_token_slot, ei)
+                    })
+                }),
+            );
+            return_message.append3(true, DbusErrorEnum::OK as u16, OK_STRING.to_string())
+        }
+        Err(e) => {
+            let (rc, rs) = engine_to_dbus_err_tuple(&e);
+            return_message.append3(default_return, rc, rs)
+        }
+    };
+    Ok(vec![msg])
+}
+
+pub fn bind_keyring(m: &MethodInfo<'_, MTSync<TData>, TData>) -> MethodResult {
+    let message: &Message = m.msg;
+    let mut iter = message.iter_init();
+    let key_desc_str: String = get_next_arg(&mut iter, 0)?;
+    let token_slot_tuple: (bool, u32) = get_next_arg(&mut iter, 1)?;
```

Comment:
Expectations for token_slot_tuple differ from published api signature which is ```(b(bu))```

Reply:
The implementation uses the correct data type. I've fixed the signature.